### DISCUSSION TITLE
order elements of foreign table on different column than id

### DISCUSSION THREAD

**Discussion by leunis** [supabase/supabase#18306] [Questions] [answered]
Hi, 
I have two tables: ' a **Product** has multiple **Series** ' And When I do this query in javascript:
```
supabase
      .from('Class')
      .select('"*",Series(*)').order('id' , {foreignTable: "Series", ascending: true } ).filter('event_id', 'eq', event_id)
```
it orders the Series on 'id'. But I want to order this collection on an other column. Is this possible with a query?

Best regards,
Martijn Dekker

**Comment by GaryAustin1** [comment, marked as answer]
You should be able to put in any column where you have 'id'.

**Reply by leunis** [reply]
I thought 'id' in my example is used for sorting 'Class'. The examples does'nt mention the use of a column on 'Series'. By default is orders the collection also on 'id'. Aha, So the sort order for Series is 'id' in my case. But how should I than also sort 'Class ' on a specific column?

**Reply by GaryAustin1** [reply]
![image](https://github.com/supabase/supabase/assets/54564956/6d3ce589-58da-47c6-986f-f76e7bfe310f)
From here: https://supabase.com/blog/postgrest-11-prerelease
This sorts the parent table...
The last line is the key part.

**Reply by leunis** [reply]
I saw this example but isn't this for a OneToOne relation and many-to-one relationships?

**Reply by GaryAustin1** [reply]
If you want to order the parent table then yes it only works if the foreign table has one matching entry.
If the foreign table has multiple entries that match then you can only use the method you started to sort those entries, not the parent.

**Reply by leunis** [reply]
I found my solution, I can chain '.order' commands.So I can do two separate 'order' commands. One for the foreign table and one for the table that gets queried. The answer I was searching for was your first anwser Thanks! Now I understand how it works
